packages/block-editor/src/components/block-list/test/block-list-context.native.js

```diff
@@ -0,0 +1,266 @@
+			findBlockLayoutByClientId,
+			updateBlocksLayouts,
+		} = DEFAULT_BLOCK_LIST_CONTEXT;
+		const currentBlockLayouts = cloneDeep( blocksLayouts );
+		const BLOCK_CLIENT_ID = PARAGRAPH_BLOCK_LAYOUT_DATA.clientId;
+
+		updateBlocksLayouts( currentBlockLayouts, PARAGRAPH_BLOCK_LAYOUT_DATA );
+
+		const findAddedBlock = findBlockLayoutByClientId(
+			currentBlockLayouts.current,
+			BLOCK_CLIENT_ID
+		);
+
+		expect( findAddedBlock ).toEqual(
+			expect.objectContaining( { clientId: BLOCK_CLIENT_ID } )
+		);
+	} );
+
+	it( "adds a new block's layout data at root level with inner blocks", () => {
+		const {
+			findBlockLayoutByClientId,
+			updateBlocksLayouts,
+		} = DEFAULT_BLOCK_LIST_CONTEXT;
+		const currentBlockLayouts = {
+			current: cloneDeep( BLOCKS_LAYOUTS_DATA ),
+		};
+		const PARENT_BLOCK_CLIENT_ID = GROUP_BLOCK_LAYOUT_DATA.clientId;
+
+		// Add parent block
+		updateBlocksLayouts( currentBlockLayouts, GROUP_BLOCK_LAYOUT_DATA );
+
+		const findAddedParentBlock = findBlockLayoutByClientId(
+			currentBlockLayouts.current,
+			PARENT_BLOCK_CLIENT_ID
+		);
+
+		expect( findAddedParentBlock ).toEqual(
+			expect.objectContaining( { clientId: PARENT_BLOCK_CLIENT_ID } )
+		);
+
+		// Add inner block to it's parent
+		updateBlocksLayouts( currentBlockLayouts, {
+			...PARAGRAPH_BLOCK_LAYOUT_DATA,
+			rootClientId: PARENT_BLOCK_CLIENT_ID,
+		} );
+
+		const findAddedInnerBlock = findBlockLayoutByClientId(
+			currentBlockLayouts.current,
+			PARAGRAPH_BLOCK_LAYOUT_DATA.clientId
+		);
+
+		expect( findAddedInnerBlock ).toEqual(
+			expect.objectContaining( {
+				clientId: PARAGRAPH_BLOCK_LAYOUT_DATA.clientId,
+			} )
+		);
+	} );
+
+	it( "adds a new block's layout data at a deep level", () => {
+		const {
+			findBlockLayoutByClientId,
+			updateBlocksLayouts,
+		} = DEFAULT_BLOCK_LIST_CONTEXT;
+		const currentBlockLayouts = {
+			current: cloneDeep( BLOCKS_LAYOUTS_DATA ),
+		};
+
+		// Add block layout data to it's parents inner blocks
+		updateBlocksLayouts( currentBlockLayouts, {
+			...PARAGRAPH_BLOCK_LAYOUT_DATA,
+			rootClientId: NESTED_WITH_INNER_BLOCKS_ID,
+		} );
+
+		const findAddedInnerBlock = findBlockLayoutByClientId(
+			currentBlockLayouts.current,
+			PARAGRAPH_BLOCK_LAYOUT_DATA.clientId
+		);
+
+		expect( findAddedInnerBlock ).toEqual(
+			expect.objectContaining( {
+				clientId: PARAGRAPH_BLOCK_LAYOUT_DATA.clientId,
+			} )
+		);
+	} );
+
+	it( "deletes a block's layout data at a root level", () => {
+		const {
+			findBlockLayoutByClientId,
+			updateBlocksLayouts,
+		} = DEFAULT_BLOCK_LIST_CONTEXT;
+		const currentBlockLayouts = {
+			current: cloneDeep( BLOCKS_LAYOUTS_DATA ),
+		};
+
+		updateBlocksLayouts( currentBlockLayouts, {
+			shouldRemove: true,
+			clientId: ROOT_LEVEL_ID,
+		} );
+
+		const findDeletedBlock = findBlockLayoutByClientId(
+			currentBlockLayouts.current,
+			ROOT_LEVEL_ID
+		);
+
+		expect( findDeletedBlock ).toEqual(
+			expect.not.objectContaining( {
+				clientId: ROOT_LEVEL_ID,
+			} )
+		);
```

Comment:
I'm wondering if we could check if the block is `null` instead of checking that doesn't contain the `clientId`, wdyt?
```suggestion
		expect( findDeletedBlock ).toBeNull();
```

Reply:
The same could be applied to the rest of the cases related to deletion.

Reply:
Good idea! Updated in https://github.com/WordPress/gutenberg/pull/39089/commits/d8f7cf8ada2d91f1233388fbd9f9953791331e37#diff-d3379351048bf4f7ab3045a566f988a3c775892c8c10449fe027c518e08709ebR229